static/cfg-view.js

```diff
@@ -0,0 +1,217 @@
+// POSSIBILITY OF SUCH DAMAGE.
+
+define(function(require){
+   "use strict";
+   
+   var FontScale = require('fontscale');
+   var options = require('options');
+   var vis = require('vis');
+   var _ = require('underscore');
+   
+    require('asm-mode');
+    require('selectize');
+    
+   
+   
+   function Cfg(hub, container, state){
+        this.container = container;
+        this.eventHub = hub.createEventHub();
+        this.domRoot = container.getElement();
+        
+        this.domRoot.html($('#cfg').html());
+        this.functions = state.cfgResult;
+        this.defaultCfgOuput = {'nodes': [{id: 0, label: 'No Output'}], 'edges': []};
+        
+        this.fnNames = this.functions? Object.keys(this.functions):  "no function";
+        this. currentFunc = this.fnNames.length? this.fnNames[0]: "";
+             
+        this.compilers = {};
+        
+        
+        this.networkOpts = {
+            autoResize: true,
+            height: '99%',
+            width: '99%',
+            locale: 'en',
+            edges: {
+                arrows: {to: {enabled: true}},
+                smooth: {enabled: true, type: 'straightCross'}
+            },
+            nodes: {
+                font: {'face': 'monospace', 'align': 'left'}
+            },
+            layout: {
+                improvedLayout: true,
+                "hierarchical": {
+                    "enabled": true,
+                    "sortMethod": "directed",
+                    "direction": "UD",
+                    nodeSpacing: 200,
+                    levelSeparation: 200
+                }
+            },
+            physics: {
+                hierarchicalRepulsion: {
+                    nodeDistance: 300
+                }
+            },
+            interaction: {
+                navigationButtons: false,
+                keyboard: {
+                    enabled: true,
+                    speed: {x: 10, y: 10, zoom: 0.03},
+                    bindToWindow: true
+                }
+            }
+
+
+        };
+        
+        this.cfgVisualiser = new vis.Network(this.domRoot.find(".graph-placeholder")[0], 
+                                             this.defaultCfgOuput, this.networkOpts);
+        
+        this.restButton = this.domRoot.find(".show-hide-btn")
+                .on('click', _.bind(function(event){
+                    this.networkOpts.interaction.navigationButtons = !this.networkOpts.interaction.navigationButtons;
+                    this.cfgVisualiser.setOptions(this.networkOpts);
+                }, this));
+        
+        this._compilerid = state.id;
+        this._compilerName = state.compilerName;
+        this._editorid = state.editorid;
+        
+        this.eventHub.on('compileResult', this.onCompileResult, this);
+        this.eventHub.on('compiler', this.onCompiler, this);
+        this.eventHub.on('compilerClose', this.onCompilerClose, this);
```

Comment:
this is only set up to update compilers...which is not used (I think)

Reply:
I am using compileResult and compiler.